receiver/receivercreator/README.md

```diff
@@ -1,12 +1,68 @@
 # receiver_creator
 
-This receiver can instantiate other receivers at runtime based on whether observed endpoints match a configured rule.
+This receiver can instantiate other receivers at runtime based on whether observed endpoints match a configured rule. These rules are evaluated for each endpoint discovered. If the rule evaluates to true then the receiver for that rule will be started against the matched endpoint.
 
-## TODO
-* Observer endpoint details
-* Rule matching
+## Config
+
+**watch_observers**
```

Comment:
Not really a suggestion for this PR as this seems to follow the style of other docs in this repo, but personally I find config docs more digestible if there's more yaml, e.g.

---

Receiver Creator will watch for endpoints generated by the extensions defined in `watch_observers`, and evaluate detected endpoints against the `rule` for each specified receiver to determine if it should be started.

```yaml
# a list of observers defined in the "extensions" section
watch_observers: [ <string> ]
# a map of receiver names to a template for when and how to instantiate that receiver
receivers:
  <string>:
    # the rule that each detected endpoint will be evaluated against to determine
    # if this receiver should be started
    rule: <string>
    # the configuration that will be used when creating the receiver
    config:
      ...
```

...